### PR TITLE
Edit ticket page

### PR DESCRIPTION
**Pull request by Val-Bustamante**
Fixed edit ticket page so that the fields are automatically filled with the values of the original ticket. Added way for users to add owners. Made deadline be of type date time when inserted into mongo.

### CODE REVIEW

**Comment by MAPReiff** [comment]
@Val-Bustamante this branch breaks the seed file:
image

**Comment by MAPReiff** [comment]
@Val-Bustamante 
Editing a ticket does not pull all of the data in, only the first word of the title/description

image
image

**Comment by MAPReiff** [comment]
@Val-Bustamante 
User `Marcus Smith` does not show up as an owner in the owners selection to make or edit a ticket. Prob need to iterate 1 more time on whatever list/array you are using for the users.

**Review by MAPReiff — COMMENTED** [review]
The changes look good!

**Comment by MAPReiff** [comment]
Github says the PR can be merged without issue, but doing a rebase with main seems to have a lot of conflicts. @Val-Bustamante @bigguyhere @dbajollari1 lets try to review the conflicts together today before I break something trying to do it myself.

**Comment by MAPReiff** [comment]
Found another issue on the PR. When editing a ticket and adding an owner, it does not update that owner’s list of owned tickets This also applies to unassigning an owner not removing the ticket from their list.

Not sure if this is for @Val-Bustamante for the page/routes/changes to the data function or for @dbajollari1 for the data functions originally.

File: data/tickets.js [hunk 1]
```diff
@@ -195,40 +197,41 @@ If we need this functionallity we can add it later */
...
   description = helpers.checkString(description, "Description");
 
-  // validate status
-  status = helpers.checkString(status, "Status");
-  if (status != "To Do" && status != "In Progress" && status != "Completed") {
-    throw new Error(
-      "status must be a string equal to, To Do, In Progress, or Completed"
-    );
-  }
+  const curTicket = await get(id);
```
Comment (bigguyhere) [inline data/tickets.js:222]: Why do we have curTicket when we have the find for the ticket on line 214? Maybe we should replace all ticket calls with curTicket

**Review by bigguyhere — APPROVED** [review]
Fixed issues with last review myself

**Comment by MAPReiff** [comment]
~~`users` can not make themselves owners of their own tickets when making a new ticket (which could be a real world use case), but `admins` can.
This works fine on editing a ticket.~~

Ignore this is propper behavior

**Review by MAPReiff — COMMENTED** [review]
Think we're good now!

**Review by MAPReiff — APPROVED** [review]

**Pull request merged**